core/schema/project.graphqls

```diff
@@ -0,0 +1,55 @@
+extend type Query {
+  "Load a project from ID."
+  project(id: ProjectID): Project!
+
+  "Load a project command from ID."
+  projectCommand(id: ProjectCommandID): ProjectCommand!
+}
+
+"A unique project identifier."
+scalar ProjectID
+
+"A unique project command identifier."
+scalar ProjectCommandID
+
+"A collection of Dagger resources that can be queried and invoked."
+type Project {
+  "A unique identifier for this project."
+  id: String!
+
+  "Initialize this project from the given directory and config path"
+  load(source: DirectoryID!, configPath: String!): Project!
+
+  "Name of the project"
+  name: String!
+
+  "Commands provided by this project"
+  commands: [ProjectCommand!]
+}
+
+"A command defined in a project that can be invoked from the CLI."
```

Comment:
Schema is missing the invocation itself, right? Are you thinking that would be a mutation, or another field in this query type?

Reply:
> Schema is missing the invocation itself, right?

Yes, invocation is through raw graphql queries right now, which are a pain to construct ([see here](https://github.com/dagger/dagger/blob/51ecf80568e15e8d34378e6406c8e66326222f4d/cmd/dagger/do.go#L168-L233))

I will probably just make this be a query rather than mutation. I think this is another case where the actual difference between the two doesn't matter to us. Leaving as a query will just be simpler since it'll work with all our existing clients.

I think the invocation api will fit nicely in with the patterns we use elsewhere of `With*`. e.g.
```go
output, err := c.Project.Load(...).
  Command("ci").Command("engine").Command("lint").
  WithFlagValue("repo", "github.com/dagger/dagger").
  WithFlagValue("branch", "main").
  Invoke(ctx)
```